delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/FilesDeletion.scala

```diff
@@ -0,0 +1,73 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.storage.files
+
+import akka.persistence.query.NoOffset
+import cats.implicits._
+import ch.epfl.bluebrain.nexus.delta.kernel.Mapper
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.files.Files.FilesAggregate
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.files.model.FileEvent
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.files.model.FileEvent._
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.Storages
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.model.StorageRejection.StorageFetchRejection
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.model.{Storage, StorageType}
+import ch.epfl.bluebrain.nexus.delta.sdk.ResourcesDeletion.{CurrentEvents, ProjectScopedResourcesDeletion, StopActor}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.Envelope
+import ch.epfl.bluebrain.nexus.delta.sdk.model.ResourcesDeletionProgress.{CachesDeleted, ResourcesDataDeleted}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.ProjectRef
+import ch.epfl.bluebrain.nexus.delta.sourcing.DatabaseCleanup
+import monix.bio.Task
+
+import java.io.File
+import scala.reflect.io.Directory
+
+final class FilesDeletion(
+    storages: Storages,
+    stopActor: StopActor,
+    currentEvents: CurrentEvents[FileEvent],
+    dbCleanup: DatabaseCleanup
+) extends ProjectScopedResourcesDeletion[FileEvent](stopActor, currentEvents, dbCleanup, Files.moduleType)(_.id) {
+
+  implicit private val mapper: Mapper[StorageFetchRejection, Throwable] =
+    rej => new IllegalArgumentException(rej.reason)
+
+  // TODO: So far we only delete files from ''DiskStorage''. To be implemented for S3Storages and RemoteDiskStorages
+  override def deleteData(projectRef: ProjectRef): Task[ResourcesDataDeleted] =
+    currentEvents(projectRef, NoOffset).flatMap { stream =>
+      stream
+        .collect {
+          case Envelope(file: FileCreated, _, _, _, _, _) if file.storageType == StorageType.DiskStorage => file.storage
+          case Envelope(file: FileUpdated, _, _, _, _, _) if file.storageType == StorageType.DiskStorage => file.storage
+        }
+        .evalMap { ref =>
+          storages.fetch(ref, projectRef).map(_.value)
+        }
+        .collect { case Storage.DiskStorage(_, _, value, _, _) => value.volume }
+        .changes
+        .evalMap { volume =>
+          Task.delay(new Directory(new File(volume.value.toFile, projectRef.toString)).deleteRecursively())
```

Comment:
This should be done at the storage level

Reply:
It is done at the files level because as soon as we start supporting S3 and remote, it will have to be done here. So it centralizes the place where the actual deletion of files happens (instead of being spread across different modules)

Reply:
For the moment, we don't know what we will do with S3 or remotes or even disk storages in other contexts than the MOOC.

Reply:
We do know that we want this to be supported in dev/staging and production, since  otherwise we would have solve the current problem just with infrastructure tricks and no code. So I think it makes sense this way. We can always move it to storages if we are not going to implement deletion for other files. 

However, If you feel strong about it I can move this logic to storages.

Reply:
It can stay as it is, we will change things later if needed with DLM